### PR TITLE
api-2 

### PR DESCRIPTION
**Pull request by manthankumbhar**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -58,6 +60,30 @@ app.post("/github_push_webhook/:id", async (req, res) => {
   }
 });
 
+app.post("/github_discord_urls", async (req, res) => {
+  try {
+    const reqBody = req.body;
+    const data = await pool.query(
```
Comment (kavadithya) [inline index.js]: make this a separate function, just like the get from db case

File: index.js [hunk 2]
```diff
@@ -58,6 +60,30 @@ app.post("/github_push_webhook/:id", async (req, res) => {
   }
 });
 
+app.post("/github_discord_urls", async (req, res) => {
+  try {
+    const reqBody = req.body;
```
Comment (kavadithya) [inline index.js:86]: check for case where url is null, and return 400

File: index.js [hunk 3]
```diff
@@ -58,6 +60,30 @@ app.post("/github_push_webhook/:id", async (req, res) => {
...
+    const data = await pool.query(
+      `insert into github_discord_url (discord_url) values ($1)`,
+      [reqBody.url]
+    );
+
+    const id = await pool.query(
```
Comment (kavadithya) [inline index.js]: use existing function

File: index.js [hunk 4]
```diff
@@ -58,6 +60,30 @@ app.post("/github_push_webhook/:id", async (req, res) => {
...
+      `select id from github_discord_url where discord_url = $1`,
+      [reqBody.url]
+    );
+
+    const id_matching_github_and_discord = id.rows[0];
+    res
+      .status(200)
+      .json(
+        `https://v2-github-discord-api-and-me.herokuapp.com/github_push_webhook/${id_matching_github_and_discord["id"]}`
```
Comment (kavadithya) [inline index.js]: construct this string outside, in a separate function. Business logic should not be inside the api code

File: index.js (hunk 3 above)
Comment (manthankumbhar) [reply index.js]: Can't, The old function is getting "discord_url" from "id" but in this case I need "id" from "discord_url"

Comment (manthankumbhar) [reply index.js]: Making a new function to get_id...discord_url

File: index.js [hunk 5]
```diff
@@ -58,6 +60,46 @@ app.post("/github_push_webhook/:id", async (req, res) => {
   }
 });
 
+async function post_discord_url_to_db() {
+  try {
```
Comment (kavadithya) [inline index.js]: remove try catch - you're swallowing the error which you should not be doing

File: index.js [hunk 6]
```diff
@@ -58,6 +60,46 @@ app.post("/github_push_webhook/:id", async (req, res) => {
   }
 });
 
+async function post_discord_url_to_db() {
```
Comment (kavadithya) [inline index.js]: rename to: insert_discord_url

File: index.js [hunk 7]
```diff
@@ -58,6 +60,46 @@ app.post("/github_push_webhook/:id", async (req, res) => {
...
+  try {
+    const data_2 = await pool.query(
+      `insert into github_discord_url (discord_url) values ($1)`,
+      [db_url]
+    );
```
Comment (kavadithya) [inline index.js]: return the row that you inserted into the db

File: index.js [hunk 8]
```diff
@@ -58,6 +60,46 @@ app.post("/github_push_webhook/:id", async (req, res) => {
...
+  }
+}
+
+async function get_id_by_discord_url() {
+  try {
+    const data_3 = await pool.query(
+      `select id from github_discord_url where discord_url = $1`,
+      [db_url]
+    );
+    const matching_id = data_3.rows[0];
```
Comment (kavadithya) [inline index.js]: variable name doesn't make sense - rename

File: index.js [hunk 9]
```diff
@@ -58,6 +60,46 @@ app.post("/github_push_webhook/:id", async (req, res) => {
...
+      `insert into github_discord_url (discord_url) values ($1)`,
+      [db_url]
+    );
+  } catch (err) {
+    console.log(err.message);
+  }
+}
+
+async function get_id_by_discord_url() {
+  try {
```
Comment (kavadithya) [inline index.js]: try block swallowing errors again

File: index.js [hunk 10]
```diff
@@ -58,6 +60,46 @@ app.post("/github_push_webhook/:id", async (req, res) => {
...
+    const data_3 = await pool.query(
+      `select id from github_discord_url where discord_url = $1`,
+      [db_url]
+    );
+    const matching_id = data_3.rows[0];
+    return matching_id["id"];
```
Comment (kavadithya) [inline index.js]: return entire row - don't return single values

File: index.js [hunk 11]
```diff
@@ -58,6 +60,46 @@ app.post("/github_push_webhook/:id", async (req, res) => {
...
+
+app.post("/github_discord_urls", async (req, res) => {
+  try {
+    const reqBody = req.body;
+    if (reqBody.url === null) {
+      return res.status(400).json({ message: "No url entered" });
+    }
+    await post_discord_url_to_db((db_url = reqBody.url));
+    const id_from_db = await get_id_by_discord_url((db_url = reqBody.url));
+    const res_url = `https://v2-github-discord-api-and-me.herokuapp.com/github_push_webhook/${id_from_db}`;
```
Comment (kavadithya) [inline index.js]: use a separate function to construct this url

File: index.js (hunk 10 above)
Comment (manthankumbhar) [reply index.js]: But I need single values only, what's the use of returning the entire row

Comment (manthankumbhar) [reply index.js:81]: Also I'm selecting the id only from the database

Comment (manthankumbhar) [reply index.js]: + if I return the entire row then how could I access the id in the function which it is used in

**Pull request closed without merging**